src/frequenz/sdk/timeseries/periodic_feature_extractor.py

```
@@ -0,0 +1,392 @@
+        MovingWindow is not considered in the profile.
+
+    Note:
+        When constructing a PeriodicFeatureExtractor object the `MovingWindow`
+        size has to be a integer multiple of the period.
+
+    Example:
+        ```python
+        from datetime import datetime, timedelta, timezone
+
+        moving_window = MovingWindow(
+            size=timedelta(days=35),
+            resampled_data_recv=microgrid.logical_meter().grid_power.new_receiver(),
+            input_sampling_period=timedelta(seconds=1),
+        )
+
+        feature_extractor = PeriodicFeatureExtractor(
+            moving_window = moving_window,
+            period=timedelta(days=7),
+        )
+
+        now = datetime.now(tzinfo=timezone.utc)
+
+        # create a daily weighted average for the next 24h
+        avg_24h = feature_extractor.avg(
+            now,
+            now + timedelta(hours=24),
+            weights=[0.1, 0.2, 0.3, 0.4]
+        )
+
+        # create a daily average for Thursday March 23 2023
+        th_avg_24h = feature_extractor.avg(datetime(2023,3,23), datetime(2023,3,24))
+        ```
+    """
+
+    def __init__(
+        self,
+        moving_window: MovingWindow,
+        period: timedelta,
+    ) -> None:
+        """
+        Initialize a PeriodicFeatureExtractor object.
+
+        Args:
+            moving_window: The MovingWindow that is used for the average calculation.
+            period: The distance between two succeeding intervals.
+
+        Raises:
+            ValueError: If the MovingWindow size is not a integer multiple of the period.
+        """
+        self._moving_window = moving_window
+        self._buffer = moving_window._buffer
+
+        self._sampling_period = self._moving_window.sampling_period.total_seconds()
+        """The sampling_period as integer to use it for indexing of samples."""
+
+        self._period = int(period.total_seconds() / self._sampling_period)
```

Does that work correctly if period is not a multiple of sampling period?

---

oh, I see you check this below. For unexpected behavior I would raise an exception or do you see valid use-cases for this?

---

I didn't want to add unnecessary restrictions and I haven't thought it fully through. I suggest we address this by adding tests in a follow up PR.